### DISCUSSION TITLE
Help translating this `bash` function for `rc.xsh`

### DISCUSSION THREAD

**Discussion by pltrz** [xonsh/xonsh#5741] [Q&A] [answered]
I'm trying to create a shell wrapper to change the current working directory when exiting [`yazi`](https://github.com/sxyazi/yazi) (tui file manager), going off the following [`bash function`](https://yazi-rs.github.io/docs/quick-start/#shell-wrapper) provided by the project (copied below, for reference).

But, something is not right. Any help appreciated in fixing (and improving) my script! Returns the error `xonsh: subprocess mode: command not found: 'cwd'`. Am I using the wrong subprocess mode?

```bash
#!/bin/bash

function y() {
    local tmp="$(mktemp -t "yazi-cwd.XXXXXX")" cwd
    yazi "$@" --cwd-file="$tmp"
    if cwd="$(command cat -- "$tmp")" && [ -n "$cwd" ] && [ "$cwd" != "$PWD" ]; then
        builtin cd -- "$cwd"
    fi
    rm -f -- "$tmp"
}
```

```xonsh
#!/bin/xonsh

@aliases.register
@aliases.return_command

def _y(args):
    """remember cwd when opening yazi"""
    tmp = $(mktemp -t)
    cwd = $(cat @(tmp))
    if (cwd != $PWD):
        return ['yazi'] + args
    else:
        rm -f $(tmp)
```

**Comment by anki-code** [comment, marked as answer]
This should work so just run it line by line and fix `rm -f $(tmp)`  to `rm -f @(tmp)`.
